ng-dev/utils/spinner.ts

```diff
@@ -7,27 +7,74 @@
  */
 
 import {cursorTo, clearLine} from 'readline';
+import {green, red} from './logging.js';
 
+/** Whether execution is in a CI environment. */
+const IS_CI = process.env['CI'];
```

Comment:
At this point in time we keep making the spinner more complicated. Should we consider just using a library here?

Reply:
This is still markedly less complicated than a library has, we can if you would like, but I do personally like the total control we have in this case.  But if you feel strongly I am happy to look into options for us.